unit_test/dda/dda.t.c

```diff
@@ -195,15 +195,15 @@ void	test_dda_perfect_sw(void)
 	r.to_cast = radian(180 + 45);
 	dda(map, start_pos, &r, &collinfo);
 	test = fpoint_to_point(collinfo.collision);
-	CU_ASSERT_EQUAL(test.x, 0);
```

Comment:
same -> +.5
test for `1, 1`

Reply:
Test returns 12, 12

Reply:
this test is perfect for the `corner_check()` flow
it fails cuz your function does not do it correctly.
in case of 225 degrees we must do -1 for both x & y